src/analytics/SegmentAnalyticsService.js

```diff
@@ -192,10 +192,18 @@ class SegmentAnalyticsService {
         resolve();
       });
 
-      // this is added to handle a specific use-case where if a user has blocked the segment
-      // domain in their browser, this promise does not get resolved and user see a blank
+      // this is added to handle a specific use-case where if a user has blocked the analytics
+      // tools in their browser, this promise does not get resolved and user see a blank
       // page. Dispatching this event in script.onerror callback in analytics.load.
       document.addEventListener('segmentFailed', resolve);
+      // This is added to hanndle the google analytics blocked case which is injected into
+      // the DOM bby segment.min.js.
+      setTimeout(() => {
+        if (!window.ga || !window.ga.create) {
+          this.segmentInitialized = false;
+          resolve();
+        }
+      }, 4000);
```

Comment:
Nit: Please use `global` instead of `window`

Otherwise, I don't have any suggestions - I was looking to you two to think through the current problem in light of our past issues.  If you're both confident, I trust your judgment.   It looks like a reasonable compromise to make sure the promise resolves. 👍🏻

Reply:
@davidjoy Addressed your suggestion, kindly take another look. Thanks!